### ISSUE TITLE
App cannot deploy on heroku-20

### ISSUE THREAD

**Issue by grenewode** [rubyforgood/mutual-aid#899]
I encountered an issue when setting up the application. I used [my fork](https://github.com/grenewode/mutual-aid) which is up to date with the most recent main (https://github.com/rubyforgood/mutual-aid/commit/5bb5a1aa575fba2c000e6e8320e137e93cc0c07b).

Once I had cloned the application locally, I used
```bash
heroku login
heroku git:remote -a APP
git push heroku main
```

as instructed by the instructions given by Heroku when creating a new app.

However, this deploying the app this way failed with the following error

```
remote:  !
remote:  !     The Ruby version you are trying to install does not exist on this stack.
remote:  !
remote:  !     You are trying to install ruby-2.7.0 on heroku-20.
remote:  !
remote:  !     Ruby ruby-2.7.0 is present on the following stacks:
remote:  !
remote:  !     - cedar-14
remote:  !     - heroku-16
remote:  !     - heroku-18
remote:  !
remote:  !     Heroku recommends you use the latest supported Ruby version listed here:
remote:  !     https://devcenter.heroku.com/articles/ruby-support#supported-runtimes
remote:  !
remote:  !     For more information on syntax for declaring a Ruby version see:
remote:  !     https://devcenter.heroku.com/articles/ruby-versions
remote:  !
```

I was able to resolve this issue by 
```bash
heroku stack:set heroku-18
```
after which the deploy seems to have worked. This was surprising though, and it would be nice if it was noted in the documentation, or if the stack could be configured by in the app itself (I don't know enough about heroku to know if this is possible).

Possibly related to #897

**Comment by sharvy** [comment]
@grenewode, At this moment Ruby 2.6.6, 2.7.2, and 3.0.0 are available on Heroku-20, which is the default stack. 

You can keep an _app.json_ file in your root folder like below, and Heroku will pick it up: 

```
{
  "stack": "heroku-18"
}
```

But, why do you want to keep Heroku specific configuration in this repo? 
What if someone wants to deploy to AWS, Google Cloud, Azure, or Digital Ocean?

**Comment by grenewode** [comment]
> @grenewode, At this moment Ruby 2.6.6, 2.7.2, and 3.0.0 are available on Heroku-20, which is the default stack.

The problem is that, until #897 gets merged, the app [requires](https://github.com/rubyforgood/mutual-aid/blob/5bb5a1aa575fba2c000e6e8320e137e93cc0c07b/Gemfile#L4) Ruby 2.7.0, rather than 2.7.2

> You can keep an _app.json_ file in your root folder like below, and Heroku will pick it up:
> 
> ```
> {
>   "stack": "heroku-18"
> }
> ```

That's cool! Thanks for the tip!

> But, why do you want to keep Heroku specific configuration in this repo?
> What if someone wants to deploy to AWS, Google Cloud, Azure, or Digital Ocean?

I agree that there probably shouldn't be Heroku specific configuration in the *code or configuration* in this repo. However, at least until #897 is merged, it might be nice to add something to the *documentation* in this repo - such as [setup.md ](https://github.com/rubyforgood/mutual-aid/blob/5bb5a1aa575fba2c000e6e8320e137e93cc0c07b/doc/setup.md#setting-up-heroku-and-using-it-for-development) or [deployment.md](https://github.com/rubyforgood/mutual-aid/blob/5bb5a1aa575fba2c000e6e8320e137e93cc0c07b/doc/deployment.md#if-you-decide-to-deploy-to-heroku)

I'd be happy to make a pull request adding a note somewhere about this but 
1. I'm not sure where to put it - maybe a new troubleshooting.md?
2. Both those files are a little sparse on info about heroku for the uninitiated (such as myself) and it might be better for someone more knowledgeable than me to write it up

**Comment by exbinary** [comment]
Thanks @grenewode and @sharvy . My instinct is to put effort into getting #782 merged instead of adding documentation that would be quickly outdated. Looks like that PR is about ready.

**Comment by exbinary** [comment]
Fixed in #782 .
